### PR TITLE
feature: powerup framework

### PR DESCRIPTION
**Pull request by Technolord3233**
added the framework for the powerup class, and made it easy for new powerups to be created. havnt actually implemented powerups to be used, this is just like the classes and stuff. for more info refer to this discord message https://discord.com/channels/1214035166008180776/1266579146293645403/1273948856735764534

### CODE REVIEW

File: server/src/model/player.ts [hunk 1]
```diff
@@ -7,6 +8,7 @@ export default class Player implements PlayerAttributes {
   public isBot: boolean;
   public isEliminated: boolean;
```
Comment (RoryT-U) [inline server/src/model/player.ts:9]: Store the power ups in the rpsMatch class instead, you can do the same as what we did for actions.
image

Just dont want to clutter up the player class with every variable we need for all the different games.

**Review by RoryT-U — COMMENTED** [review]
yeah besides changing where the power ups are stored, its all good.

File: server/src/model/powerups/powerup.ts
Comment (lkog77) [inline server/src/model/powerups/powerup.ts:1]: Looks great, not too sure if the targetSelf and instantUse variables are required I guess it depends how we implement the powerups with the socketio backend.

**Review by lkog77 — APPROVED** [review]
Overall looks good.

**Pull request merged**